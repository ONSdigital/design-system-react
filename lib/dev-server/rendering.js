import fs from "fs";

const DEFAULT_PARAMS = {
  bodyClass: "",
};

export function renderPage(params) {
  const templatePath = "./lib/dev-server/template.html";
  const template = fs.readFileSync(templatePath, { encoding: "utf8" });

  params = Object.assign({}, DEFAULT_PARAMS, params);

  let result = template;
  for (let paramName of Object.keys(params)) {
    result = result.replace(new RegExp(`[{][{]${paramName}[}][}]`, "g"), params[paramName]);
  }

  return result;
}
