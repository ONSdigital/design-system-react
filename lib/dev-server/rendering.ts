import fs from "fs-extra";

const DEFAULT_PARAMS = {
  designSystemVersion: "1.2.3",
  bodyClass: "",
};

export function renderPage(params: any): string {
  const templatePath = "./lib/dev-server/template.html";
  const template = fs.readFileSync(templatePath, { encoding: "utf8" });

  params = Object.assign({}, DEFAULT_PARAMS, params);

  let result = template;
  for (let paramName of Object.keys(params)) {
    result = result.replace(new RegExp(`[{][{]${paramName}[}][}]`, "g"), params[paramName]);
  }

  return result;
}
