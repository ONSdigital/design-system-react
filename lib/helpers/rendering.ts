import fs from "fs-extra";

const DEFAULT_PARAMS = {
  designSystemVersion: "1.2.3",
};

export function loadTemplate(templatePath: string): string {
  return fs.readFileSync(templatePath, { encoding: "utf8" });
}

export function renderPage(template: string, params: any): string {
  params = Object.assign({}, DEFAULT_PARAMS, params);

  let result = template;
  for (let paramName of Object.keys(params)) {
    result = result.replace(new RegExp(`[{][{]${paramName}[}][}]`, "g"), params[paramName]);
  }

  return result;
}
