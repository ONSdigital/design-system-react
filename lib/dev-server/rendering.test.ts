import { renderPage } from "./rendering";

describe("dev-server/rendering", () => {
  describe("renderPage(params)", () => {
    it("assumes the default body class of ''", () => {
      const html = renderPage({});

      expect(html).toMatchSnapshot();
    });

    it("substitutes placeholder values", () => {
      const html = renderPage({
        designSystemVersion: "1.2.3",
        title: "An example page",
        bodyClass: "an-example-body-class",
        body: `
          <p>Example body content.</p>
        `,
      });

      expect(html).toMatchSnapshot();
    });
  });
});
