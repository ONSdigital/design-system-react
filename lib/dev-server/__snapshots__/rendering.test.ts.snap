// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dev-server/rendering renderPage(params) assumes the default body class of '' 1`] = `
"<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <title>{{title}} - Design System React</title>
    <link rel="stylesheet" href="https://cdn.ons.gov.uk/sdc/design-system/61.0.4/css/main.css">
    <link rel="stylesheet" href="/main.css">
  </head>
  <body class="">
    {{body}}
  </body>
</html>
"
`;

exports[`dev-server/rendering renderPage(params) substitutes placeholder values 1`] = `
"<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <title>An example page - Design System React</title>
    <link rel="stylesheet" href="https://cdn.ons.gov.uk/sdc/design-system/61.0.4/css/main.css">
    <link rel="stylesheet" href="/main.css">
  </head>
  <body class="an-example-body-class">
    
          <p>Example body content.</p>
        
  </body>
</html>
"
`;
